diva-irma.js

```diff
@@ -236,54 +171,33 @@ function startIssueSession(credentials, attributes, attributeLabel) {
  * Decode and verify JWT verify token from api server and check validity/signature
  * @function verifyIrmaJwt
  * @param {string} token JWT string
+ * @param {object} jwtOptions JWT verification options
  * @returns {Promise<json>} decoded IRMA JWT token from api server
  */
-function verifyIrmaApiServerJwt(token) {
+function verifyIrmaApiServerJwt(token, jwtOptions) {
   const key = divaConfig.irmaApiServerPublicKey;
-  return BPromise.try(() => jwt.verify(token, key, divaConfig.jwtIrmaApiServerVerifyOptions));
-}
-
-function verifyIrmaApiServerSignature(token) {
-  const key = divaConfig.irmaApiServerPublicKey;
-  return BPromise.try(() => jwt.verify(token, key, divaConfig.jwtIrmaApiServerSignatureOptions));
-}
-
-function addIrmaProof(proofResult, irmaSessionId) {
-  const divaSessionId = proofResult.jti;
-
-  return divaState.getDivaEntry(divaSessionId)
-    .then(divaStateEntry =>
-      divaState.setDivaEntry(divaSessionId, {
-        ...divaStateEntry,
-        [irmaSessionId]: proofResult,
-      }),
-    );
+  return BPromise.try(() => jwt.verify(token, key, jwtOptions));
 }
 
 function completeDisclosureSession(irmaSessionId, token) {
-  return verifyIrmaApiServerJwt(token)
-    .then(proofResult => addIrmaProof(proofResult, irmaSessionId))
-    .then(() => divaState.setIrmaEntry(irmaSessionId, 'COMPLETED'));
+  return verifyIrmaApiServerJwt(token, divaConfig.jwtIrmaApiServerVerifyOptions)
+    .then((disclosureProofResult) => {
+      divaState.setIrmaEntry(irmaSessionId, 'COMPLETED'); // Async
+      // TODO: .then(proofResult => addIrmaProof(proofResult, irmaSessionId))
```

Comment:
Why is this TODO introduced? Because this is now something that needs to be fixed at the caller side?

Reply:
I shouldn't have comitted that line, I'll remove it!

But yes, this should indeed be fixed at the caller side.